server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -207,6 +211,12 @@ public Map<RepositoryShardId, Set<ShardGeneration>> obsoleteGenerations(String r
                     // We moved from a non-null generation successful generation to a different non-null successful generation
                     // so the original generation is clearly obsolete because it was in-flight before and is now unreferenced everywhere.
                     obsoleteGenerations.computeIfAbsent(repositoryShardId, ignored -> new HashSet<>()).add(oldStatus.generation());
+                    logger.debug(
+                        "Marking shard generation [{}] file for cleanup. The finalized shard generation is now [{}] for shard snapshot [{}]",
```

Comment:
I think we should include the shard ID here too

Reply:
I don't see a way to get the Shard ID here. Repository ID is simple. But it looks like the InProgressSnapshots.Entry has only a shards map of ShardID->ShardSnapshotStatus, which doesn't do much good here.

Reply:
Nvm, repositoryShardId.shardId() is the ShardID. I was confusing things with shard generation, again.